webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/EditorTechnique.scala

```diff
@@ -323,6 +333,30 @@ class TechniqueSerializer(parameterTypeService: ParameterTypeService) {
       )
     }
 
+    def serializeMethod(method : MethodElem): JValue = {
+      method match {
+        case c : MethodCall => serializeMethodCall(c)
+        case b : MethodBlock => serializeMethodBlock(b)
+      }
+    }
+
+    def serializeCompositionRule(reportingLogic: ReportingLogic) :JValue = {
+      import ReportingLogic._
+      reportingLogic match {
+        case WorstReport =>  ("type"-> "worst")
+        case SumReport =>  ("type"-> "sum")
+        case FocusReport(component)  => ("type"  -> "focus") ~ ("value" -> component)
```

Comment:
ok, so you need to split `name` and `serialize` in `ReportingLogic` (with by default `val serialize = name` and for focus report something different)